### PR TITLE
Native: extend CryptoLib's `verifyWithECDsa` with compatible `NamedCurveHash` argument

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Port protocol-related part of https://github.com/neo-project/neo/pull/3209.

Note, that build and tests are failing because we need to update `neo` git submodule. We can merge https://github.com/neo-project/neo/pull/3209 firstly, and then update the `neo` submodule onto the fresh master. Then I'll rebase this PR and check that tests are passed.

### CODE REVIEW

**Comment by Jim8y** [comment]
@AnnaShaleva anna, we are not going to add secp256k1 support here. Core only do verification, no construction.

**Comment by AnnaShaleva** [comment]
This code is needed for verification only. We extend the native CryptoLib's `verifyWithECDsa` parameter, so that both Sha256 and Keccak256 hashes may be checked. We need this PR.

**Review by Jim8y — APPROVED** [review]

**Review by vncoelho — COMMENTED** [review]
I will wait until the dependent PR is merged.

File: src/Neo.SmartContract.Framework/Native/NamedCurveHash.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+        /// <summary>
+        /// The secp256k1 curve and Keccak256 hash algorithm.
+        /// </summary>
+        secp256k1Keccak256 = 122,
+
+        /// <summary>
+        /// The secp256r1 curve, which known as prime256v1 or nistP-256, and Keccak256 hash algorithm.
+        /// </summary>
+        secp256r1Keccak256 = 123
```
Comment (shargon) [inline src/Neo.SmartContract.Framework/Native/NamedCurveHash.cs:39]: Why don't just add these values to the previous enum?

Comment (cschuchardt88) [reply src/Neo.SmartContract.Framework/Native/NamedCurveHash.cs:39]: Hash isn't related to a/the curve. So I don't know why you need `secp256k1` in the name. `Keccak-256` is a hash function. If we decide to support on `secp256r1` curves. Than you would have to add `secp256r1SHA256`, `secp256k1Keccak256`, etc; should only be two values `Keccak256` and `SHA256` for `hasher enum`. Why do we need all this extra stuff?

If it's to indicate a `prefix` within the hash than a more appropriate name should be used.

**Review by Jim8y — DISMISSED** [review]

**Comment by AnnaShaleva** [comment]
Can’t be merged, I need to update dependencies.

**Comment by vncoelho** [comment]
I imagined that, fetch git as well

**Comment by AnnaShaleva** [comment]
Updated, ready for review and merge.

**Review by cschuchardt88 — APPROVED** [review]

**Review by Jim8y — APPROVED** [review]

**Pull request merged**